bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ArrayValueImpl.java

```diff
@@ -533,6 +543,73 @@ public void add(long index, BString value) {
         addBString(index, value);
     }
 
+    public void addRefValueForcefully(int index, Object value) {
+        switch (this.elementReferredType.getTag()) {
+            case TypeTags.BOOLEAN_TAG:
```

Comment:
And these lines too

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/42449 to address this.